### PR TITLE
[Confluence] Allow to ingest space separately

### PR DESCRIPTION
**Pull request by parth-elastic**
## Closes https://github.com/elastic/connectors-py/issues/2578

This PR allows to ingest each space separately.

Performance stats with the updated code:
- https://docs.google.com/document/d/1EBtLn1nttBsxBNxpXDTBfACAqHj1oVacdfK6LvmAVjo/edit 

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference
- [ ] if you added or changed Rich Configurable Fields for a Native Connector, you made a corresponding PR in [Kibana](https://github.com/elastic/kibana/blob/main/packages/kbn-search-connectors/types/native_connectors.ts)

#### Changes Requiring Extra Attention

- [ ] Security-related changes (encryption, TLS, SSRF, etc)
- [ ] New external service dependencies added.

### CODE REVIEW

**Comment by moxarth-elastic** [comment]
buildkite test this

**Comment by moxarth-elastic** [comment]
buildkite test this

**Comment by navarone-feekery** [comment]
@moxarth-elastic can you resolve the CI problems? I can review once it's green.

**Comment by moxarth-elastic** [comment]
> @moxarth-elastic can you resolve the CI problems? I can review once it's green.

Yes, looking into it.

**Comment by moxarth-elastic** [comment]
buildkite test this

File: tests/sources/fixtures/confluence/fixture.py [hunk 1]
```diff
@@ -20,16 +21,16 @@
...
-        SPACE_COUNT = 100
-        SPACE_OBJECT_COUNT = 200
+        SPACE_COUNT = 10
+        SPACE_OBJECT_COUNT = 50
         ATTACHMENT_COUNT = 5
     case "large":
-        SPACE_COUNT = 100
-        SPACE_OBJECT_COUNT = 250
+        SPACE_COUNT = 10
+        SPACE_OBJECT_COUNT = 75
```
Comment (navarone-feekery) [inline tests/sources/fixtures/confluence/fixture.py:33]: This looks to me like a significant reduction in performance. Can you perform a benchmark comparison for this PR against the current Confluence connector?
If the performance hit is significant, we might want to consider making this optional through an RCF.

Comment (moxarth-elastic) [reply tests/sources/fixtures/confluence/fixture.py:33]: Now we are indexing the documents space by space

If we calculate the number of docs to be indexed in functional test, the stats are as follows for the `medium` DATA_SIZE:
fixture with the older approach where spaces and other objects are getting indexed separately:
- `SPACE_COUNT + (SPACE_OBJECT_COUNT * ATTACHMENT_COUNT * 2)`
- Total docs - `100 + 200 * 5 * 2 = 2100`

whereas in updated fixture:
- `SPACE_COUNT + (SPACE_COUNT * SPACE_OBJECT_COUNT * ATTACHMENT_COUNT * 2)`
- Total docs - `10 + 10 * 50 * 5 * 2 = 5010`

So, eventually ftest is indexing more docs now.

Comment (moxarth-elastic) [reply tests/sources/fixtures/confluence/fixture.py:33]: If we don't reduce the numbers in the fixture, it'll try to index `100 (spaces) + 100 (spaces) * 200 (page/blogpost) * 2 * 5 (attachment) = 2,00,010 doc` and so the CI will fail.

Comment (moxarth-elastic) [reply tests/sources/fixtures/confluence/fixture.py:33]: @navarone-feekery we're preparing a performance doc, will attach it in the description.

File: connectors/sources/confluence.py [hunk 2]
```diff
@@ -1236,32 +1235,26 @@ async def get_docs(self, filtering=None):
...
-            await self.fetchers.put(self._space_coro)
-            await self.fetchers.put(
-                partial(
-                    self._page_blog_coro,
-                    f"{configured_spaces_query}{BLOGPOST}&{CONTENT_QUERY}",
-                    BLOGPOST,
+            async for space in self._space_coro():
+                yield space, None
+                self._logger.info(f"Fetching docs from space: {space['key']}")
+                configured_spaces_query = f"cql=space in ('{space['key']}') AND type="
```
Comment (navarone-feekery) [inline connectors/sources/confluence.py:1241]: I think it would be valuable to have this query and the ones below added to debug logs.

**Review by navarone-feekery — COMMENTED** [review]
Looks good! Just one change request regarding logging.

File: tests/sources/fixtures/confluence/fixture.py (hunk 1 above)
Comment (navarone-feekery) [reply tests/sources/fixtures/confluence/fixture.py:33]: Thanks for the extra effort here, I'm no longer concerned about the performance impact.

File: connectors/sources/confluence.py (hunk 2 above)
Comment (moxarth-elastic) [reply connectors/sources/confluence.py:1244]: Added a logger for the CQL in `_page_blog_coro`

**Review by navarone-feekery — APPROVED** [review]

**Pull request merged**